### PR TITLE
Fix/make ap (almost) work outside the dmt repo

### LINKED ISSUE #4: refactor(analysis-platform): move to a separate repo called dm-app-analysis-platform
For testing how easy/hard it is to create custom applications.

- [x] Copied over AP code from dmt repo to https://github.com/equinor/dm-app-analysis-platform
- [ ] #3

### LINKED ISSUE #5: refactor (dm-app-analysis-platform): update to make the portal stuff identical as create-dm-app
make App.tsx have same format as the https://github.com/equinor/create-dm-app.

Use the same structure as introduced with https://github.com/equinor/create-dm-app/commit/2252c373f10aadd003960478da475040546f9f7f and https://github.com/equinor/create-dm-app/commit/a67901cb15c25f1b1dbee707e9411840e8373aec (including settings file)

### PR DESCRIPTION
**Pull request by KristianKjerstad**
## What does this pull request change?
- various bug fixes
- import code from dmt repo
- load appMooring and sima blueprints + entities into the apps/ folder
- linting
- add plugins: task, sima-application-input, job-handlers

A lot of code changes are due to copying of code from the dmt repo
## Why is this pull request needed?
do necessary setup and code changes in the analysis platform code to make the app work outside the dmt repo

NB! due to a bug with cli, all datasources are not included. This must be included after bug with relative imports in dm cli are fixed 

## Issues related to this change
closes #5 
closes #4

### CODE REVIEW

**Comment by KristianKjerstad** [comment]
closes #4

**Comment by KristianKjerstad** [comment]
blocked by https://github.com/equinor/dm-core-packages/pull/16/files

after that is merged, this pr can update package versions to the latest

File: src/App.tsx [hunk 1]
```diff
@@ -47,132 +49,132 @@ const HorizontalList = styled.div`
...
+  const { applications } = props
+  const history = useHistory()
+  const links = Object.values(applications).map((setting) => (
+    <div key={setting.name}>
+      <CardWrapper onClick={() => history.push(`/${setting.urlPath}`)}>
+        <CardHeader>
+          <CardHeading>{`${setting.label}`}</CardHeading>
+        </CardHeader>
+        <CardBody>
+          <CardFieldset>{`${setting.description}`}</CardFieldset>
```
Comment (soofstad) [inline src/App.tsx:62]: Think this should follow the "new bootstrapper standard" now. Meaning no "portal stuff

File: apps/sima/settings.json [hunk 2]
```diff
@@ -0,0 +1,11 @@
+{
+  "type": "system/SIMOS/Application",
+  "name": "sima",
+  "description": "",
+  "label": "SIMA Blueprints",
+  "tabIndex": 3,
+  "urlPath": "sima",
+  "hidden": false,
+  "visibleDataSources": [ "sima"],
```
Comment (netr0m) [inline apps/sima/settings.json:9]:
```suggestion
  "visibleDataSources": ["sima"],
```

File: apps/sima/settings.json [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+  "packages": [ "sima/input", "sima/marmo"]
```
Comment (netr0m) [inline apps/sima/settings.json:10]:
```suggestion
  "packages": ["sima/input", "sima/marmo"]
```

File: src/App.tsx [hunk 4]
```diff
@@ -47,132 +49,132 @@ const HorizontalList = styled.div`
...
+  ))
+  return (
+    <div>
+      <HorizontalList>{links}</HorizontalList>
+    </div>
+  )
 }
 
-const dataSourceId = "DemoDS"
+const dataSourceId = 'DemoDS'
```
Comment (netr0m) [inline src/App.tsx:74]:
```suggestion
const dataSourceId = 'DemoDS'
```
should probably be read from `app-settings.json` (as in `create-dm-app`):
- https://github.com/equinor/create-dm-app/blob/main/src/App.tsx#L14
- https://github.com/equinor/create-dm-app/blob/main/src/App.tsx#L33-L35
- https://github.com/equinor/create-dm-app/blob/main/src/app-settings.json

**Review by netr0m — APPROVED** [review]
A few remarks

File: app/analysis-platform/data/AnalysisPlatformDS/models/AnalysisPlatform.json [hunk 5]
```diff
@@ -0,0 +1,28 @@
+{
```
Comment (soofstad) [inline app/analysis-platform/data/AnalysisPlatformDS/models/AnalysisPlatform.json:1]: This file might want the "visibleDataSource" attribute

**Review by soofstad — CHANGES_REQUESTED** [review]

File: app/analysis-platform/data/AnalysisPlatformDS/instances/AnalysisPlatform.json [hunk 6]
```diff
@@ -2,5 +2,6 @@
   "_id": "4483c9b0-d505-46c9-a157-94c79f4d7b7b",
   "type": "AnalysisPlatformDS/models/AnalysisPlatform",
   "label": "Analysis Platform App",
-  "urlPath": ""
+  "urlPath": "ap",
```
Comment (netr0m) [inline app/analysis-platform/data/AnalysisPlatformDS/instances/AnalysisPlatform.json:5]:
```suggestion
```
I believe the `urlPath` property is obsolete now. @soofstad ?

File: src/plugins/analysis-platform/App.tsx [hunk 7]
```diff
@@ -34,24 +35,31 @@ export default (props: TApp): JSX.Element => {
   const settings = useContext(ApplicationContext)
   const appConfig = useContext(ApplicationContext)
 
-  const urlPath = settings.urlPath ? `/${settings.urlPath}` : ''
-  console.log('appConfig', appConfig)
+
+    plan:
+    få til routing å fungere. Drit i andre datasources, siden dm-cli ikke funker med reset.
+    Av en eller annen grunn vil ikke komponenter definert i Routes rendre når man går til riktig url.
```
Comment (netr0m) [inline src/plugins/analysis-platform/App.tsx:41]:
```suggestion
```
Accidental paste?

**Review by netr0m — CHANGES_REQUESTED** [review]
See comments

File: app/analysis-platform/data/AnalysisPlatformDS/instances/AnalysisPlatform.json (hunk 6 above)
Comment (KristianKjerstad) [reply app/analysis-platform/data/AnalysisPlatformDS/instances/AnalysisPlatform.json:5]: good point!
will remove urlpath and the /ap url prefix from the app

**Review by soofstad — APPROVED** [review]

**Pull request merged**